waku/handshake.go

```diff
@@ -158,6 +206,58 @@ loop:
 	return s.ListEnd()
 }
 
+func (o *statusOptions) addKeyToType(ktt *statusOptionKeyToType) {
+
+	if o.keyTypeMapping.idxFieldKey == nil {
+		o.keyTypeMapping.idxFieldKey = make(map[int]*statusOptionKeyToType)
+	}
+
+	if o.keyTypeMapping.keyFieldIdx == nil {
+		o.keyTypeMapping.keyFieldIdx = make(map[statusOptionKey]*statusOptionKeyToType)
+	}
+
+	o.keyTypeMapping.idxFieldKey[ktt.Idx] = ktt
+	o.keyTypeMapping.keyFieldIdx[ktt.Key] = ktt
+}
+
+func (k statusOptionKeyToType) decodeStream(s *rlp.Stream) error {
```

Comment:
```suggestion
func (k *statusOptionKeyToType) decodeStream(s *rlp.Stream) error {
```

Shouldn't it be a pointer receiver? You assign values to `k` below.

Reply:
Ah, yes. Thank you. I can't believe I missed this.